src/libfsm/print/ir.h

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2018 Katherine Flavel
+ *
+ * See LICENCE for the full copyright terms.
+ */
+
+#ifndef FSM_INTERNAL_IR_H
+#define FSM_INTERNAL_IR_H
+
+/*
+ * Codegen IR intended for C-like switch statements
+ *
+ * The fsm_state structs don't exist from this IR onwards.
+ * These nodes are always DFA. No epsilons, No duplicate edge labels,
+ * so this IR takes advantage of that.
+ *
+ * But this is not neccessarily a minimal DFA. By this stage,
+ * we're done with graph algorithmics.
+ */
+
+enum ir_strategy {
+	IR_NONE     = 1 << 0,
```

Comment:
Based on the comment below, is this simultaneously being used as a type tag for the `struct ir_state` union, but also as a set of allowed strategies in `make_ir`?

Reply:
No, just the former. I was originally thinking of adding a mask of which strategies to allow, but I tried that and didn't like it much, and currently I think it'd make more sense to have options like "always make a table" set in `struct fsm_options` instead.